velox/functions/prestosql/tests/StringFunctionsTest.cpp

```diff
@@ -1457,6 +1461,124 @@ void StringFunctionsTest::testReplaceFlatVector(
+  testReplaceFirstFlatVector(
+      {{{"Testcases test cases", "cases", ""}, {"Test test cases"}}});
+  testReplaceFirstFlatVector(
+      {{{"test cases", "", "Add "}, {"Add test cases"}}});
+  testReplaceFirstFlatVector({{{"", "", "not_empty"}, {"not_empty"}}});
+  testReplaceFirstFlatVector({{{"", "foo", "bar"}, {""}}});
+
+  auto testReplaceFirstInPlace =
+      [this](
+          const std::vector<std::pair<std::string, std::string>>& tests,
+          const std::string& search,
+          const std::string& replace,
+          bool multiReferenced) {
+        auto makeInput = [&]() {
+          auto stringVector = makeFlatVector<StringView>(tests.size());
+
+          for (int i = 0; i < tests.size(); i++) {
+            stringVector->set(i, StringView(tests[i].first));
+          }
+          auto crossRefVector = makeFlatVector<StringView>(1);
+
+          if (multiReferenced) {
+            crossRefVector->acquireSharedStringBuffers(stringVector.get());
+          }
+          return stringVector;
+        };
+
+        auto testResults = [&](const FlatVector<StringView>* results) {
+          for (int32_t i = 0; i < tests.size(); ++i) {
+            ASSERT_EQ(results->valueAt(i), StringView(tests[i].second));
+          }
+        };
+
+        auto result = evaluate<FlatVector<StringView>>(
+            fmt::format("replaceFirst(c0, '{}', '{}')", search, replace),
+            makeRowVector({makeInput()}));
+        testResults(result.get());
+
+        // Test in place optimization. If in-place is expected, make sure it
+        // happened. If its not expected make sure it did not happen.
+        auto applyReplaceFirstFunction =
+            [&](std::vector<VectorPtr>& functionInputs, VectorPtr& resultPtr) {
+              core::QueryConfig config({});
+              auto replaceFunction = exec::getVectorFunction(
+                  "replaceFirst",
+                  {VARCHAR(), VARCHAR(), VARCHAR()},
+                  {},
+                  config);
+              SelectivityVector rows(tests.size());
+              ExprSet exprSet({}, &execCtx_);
+              RowVectorPtr inputRows = makeRowVector({});
+              exec::EvalCtx evalCtx(&execCtx_, &exprSet, inputRows.get());
+              replaceFunction->apply(
+                  rows, functionInputs, VARCHAR(), evalCtx, resultPtr);
+            };
+
+        std::vector<VectorPtr> functionInputs = {
+            makeInput(),
+            makeConstant(search.c_str(), tests.size()),
+            makeConstant(replace.c_str(), tests.size())};
+
+        VectorPtr resultPtr;
+        applyReplaceFirstFunction(functionInputs, resultPtr);
+        testResults(resultPtr->asFlatVector<StringView>());
+
+        if (!multiReferenced && search >= replace) {
+          // Expected in-place.
+          ASSERT_TRUE(resultPtr == functionInputs[0]);
+        } else {
+          ASSERT_FALSE(resultPtr == functionInputs[0]);
+        }
+      };
```

Comment:
Just noticed this, you should be able to re-use `StringFunctionsTest::testReplaceInPlace` by adding another input param `functionName` then using it instead of the hardcoded `replace` in its implementation. The test logic is fairly involved here, so would be good to keep it shared.